### PR TITLE
MBS-12346 Create draft of the new Emcee Android Worker

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

File: subprojects/emcee/android-device/src/main/kotlin/com/avito/emcee/device/AndroidDevice.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.avito.emcee.device
+
+public interface AndroidDevice {
+    public suspend fun installApplication(application: AndroidApplication): Result<Unit>
+    public suspend fun checkIsAlive(): Boolean
```
Comment (eugene-krivobokov) [inline subprojects/emcee/android-device/src/main/kotlin/com/avito/emcee/device/AndroidDevice.kt]: How about renaming it to `isAlive()`.
I think "check" adds only noise to API and can be omitted: "checkX", "checkY", "checkZ"

File: subprojects/emcee/android-device/src/main/kotlin/com/avito/emcee/device/internal/Emulator.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.avito.emcee.device.internal
+
+import com.avito.emcee.device.AndroidApplication
+import com.avito.emcee.device.AndroidDevice
+
+internal class Emulator : AndroidDevice {
```
Comment (eugene-krivobokov) [inline subprojects/emcee/android-device/src/main/kotlin/com/avito/emcee/device/internal/Emulator.kt]: "emulator" looks like a leaky abstraction.
Should we know here about the implementation? Does a real device connected by adb differ?

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/StartWorkerCommand.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@ExperimentalTime
+@ExperimentalCli
+internal class StartWorkerCommand(
+    name: String,
+    description: String
+) : Subcommand(name, description) {
+
+    private val workerId: String by option(
+        type = ArgType.String,
+        description = "Help Queue to identify worker's interactions"
```
Comment (eugene-krivobokov) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/StartWorkerCommand.kt]:
```suggestion
        description = "Helps Queue to identify worker's interactions"
```
Is it a statement or request?

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/emcee/android-device/src/main/kotlin/com/avito/emcee/device/AndroidDevice.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.avito.emcee.device
+
+public interface AndroidDevice {
+    public suspend fun installApplication(application: AndroidApplication): Result<Unit>
```
Comment (dsvoronin) [inline subprojects/emcee/android-device/src/main/kotlin/com/avito/emcee/device/AndroidDevice.kt]: Should be mirrored with `uninstall` fun, i propose to remove `Application` from name here

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/AndroidDeviceState.kt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            }
+            return Clean(device)
+        }
+
+        override fun testExecuted(): AndroidDeviceState {
+            return WithDiskCache(device, apps)
+        }
+    }
+
+    class WithDiskCache(
```
Comment (dsvoronin) [inline subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/AndroidDeviceState.kt]: i don't understand this state's purpose from that name, can you please explain?

**Review by dsvoronin — APPROVED** [review]

File: subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/AndroidDeviceState.kt (hunk 5 above)
Comment (sergioboi) [reply subprojects/emcee/worker/src/main/kotlin/com/avito/emcee/worker/internal/AndroidDeviceState.kt]: This `WithDiskCache ` means state when a test leaves data on the device disk. 
I could rename it to `WithDataOnDisk`

**Pull request merged**